DashAI/back/plugins/utils.py

```diff
@@ -68,3 +76,79 @@ def get_plugins_from_pypi() -> List[dict]:
         if plugin_name.lower().startswith("dashai") and plugin_name.lower() != "dashai"
     ]
     return [_get_plugin_by_name_from_pypi(plugin_name) for plugin_name in plugins_names]
+
+
+def available_plugins() -> List[type]:
+    """
+    Get available DashAI plugins entrypoints
+
+    Returns
+    ----------
+    List[type]
+        A list of plugins' classes
+    """
+    # Retrieve plugins groups (DashAI components)
+    plugins = entry_points(group="dashai.plugins")
+
+    # Look for installed plugins
+    plugins_list = []
+    for plugin in plugins:
+        # Retrieve plugin class
+        plugin_class = plugin.load()
+        plugins_list.append(plugin_class)
+
+    return plugins_list
+
+
+def register_new_plugins(component_registry) -> List[type]:
+    """
+    Register only new plugins in component registry
+
+    Parameters
+    ----------
+    component_registry : ComponentRegistry
+        The current app component registry
+
+    Returns
+    ----------
+    List[type]
+        A list of plugins' classes that were registered in the component registry
+    """
+    installed_plugins = []
+    new_plugins = []
+    for component in component_registry.get_components_by_types():
+        installed_plugins.append(component["class"])
+    for plugin in available_plugins():
+        if plugin not in installed_plugins:
+            new_plugins.append(plugin)
+            # The component shouldnt be registered if it does not inherit from
+            # any DashAI base class with a 'TYPE' class attribute.
+            with contextlib.suppress(TypeError):
+                component_registry.register_component(plugin)
```

Comment:
Here I would suggest using a try/catch + logger.info or logger.exception to notify that the plug-in could not be installed.

Reply:
Fixed!